parser.cpp

```diff
@@ -732,6 +804,43 @@ Dims* Parser::parse_dims()
     return dims;
 }
 
+void Parser::parse_directive()
+{
+    if (peek() == "ifdef")
+    {
+        next();
+        Token t = next();
+        std::string name = static_cast<std::string>(t);
+        if (!pp_.process(Ifdef, name))
+            ERROR("unexpected error with #ifdef");
+    }
+    else if (peek() == "ifndef")
+    {
+        next();
+        Token t = next();
+        std::string name = static_cast<std::string>(t);
+        if (!pp_.process(Ifndef, name))
+            ERROR("unexpected error with #ifndef");
+    }
+    else if (peek() == "else")
```

Comment:
Would it be possible to support `elseif`

Reply:
`elseif` only makes sense when we support `if`. We can consider support later if needed.

Reply:
Ah ok.